controller/src/main/java/io/pravega/controller/store/stream/ZKStream.java

```diff
@@ -179,6 +182,21 @@ public void refresh() {
                 });
     }
 
+    @Override
+    CompletableFuture<Void> createSealedSegmentsRecord(byte[] sealedSegmentsRecord) {
+        return store.createZNodeIfNotExist(sealedSegmentsPath, sealedSegmentsRecord);
```

Comment:
If I understand it right, we are adding one node per stream, is that it? What's the current estimate of metadata per stream we are storing in zk?

Reply:
yes.
basically we have following nodes per stream:
1. segment table (size is function of number of segments)
2. history table (size is function of number of scale operations)
3. index table (size is function of number of scale operations)
4. TruncationRecord (size is function of number of deleted segments)
5. SealedSegmentsRecord (size is function of number of sealed segments)
6. configuration (constant size)
7. state (constant size)
8. transaction records (active txns, completed txns).

Reply:
Ok, I have created a wiki page to keep track of it:

https://github.com/pravega/pravega/wiki/ZooKeeper-Data-Layout

Future changes to the layout should be reflected on that page.